mrT.go

```diff
@@ -415,79 +483,86 @@ func (m *MrT) ForEachTxn(txnID string, archive bool, fn ForEachTxnFn) (err error
 
 // LastTxn will get the last transaction id
 func (m *MrT) LastTxn() (txnID string, err error) {
-	m.mux.Lock()
-	defer m.mux.Unlock()
-	if m.closed {
-		return "", errors.ErrIsClosed
+	if m.closed.Get() {
+		err = errors.ErrIsClosed
+		return
 	}
-	defer m.s.SeekToEnd()
-	return peekLastTxn(m.s)
+
+	txnID = m.ltxn.Load()
+	return
 }
 
 // Archive will archive the current data
 func (m *MrT) Archive(populate TxnFn) (err error) {
-	var (
-		af      file.Interface
-		txn     Txn
-		lastTxn string
-	)
+	if err = m.f.With(func(f *os.File) (err error) {
+		if m.closed.Get() {
+			return errors.ErrIsClosed
+		}
 
-	m.mux.Lock()
-	defer m.mux.Unlock()
-	if m.closed {
-		return errors.ErrIsClosed
-	}
+		aw := m.af.Writer()
+		defer aw.Close()
 
-	if lastTxn, err = peekLastTxn(m.s); err != nil {
-		return
-	}
+		// Ensure archive file is at the end
+		if _, err = aw.Seek(0, os.SEEK_END); err != nil {
```

Comment:
can we use the new io.SeekEnd, etc constants? the os ones are deprecated.

Reply:
Oh sure!